### PR TITLE
Issue #4158 Re-enable support for duplicate session cookies.

### PR DESCRIPTION
**Pull request by janbartel**
#4158 

Reinstate behaviour of examining more than one session cookie in a request to find a valid one. Add new behaviour: if there is more than one valid session cookie, error 400 is returned.

### CODE REVIEW

**Comment by joakime** [comment]
This change broke many tests.

**Comment by janbartel** [comment]
There's only  a couple of tests in error, but the same tests in multiple session flavours.

The problem is with distinguishing the requested session id vs obtaining a valid session from the id, when there is more than one session id. The requested session id does not need to be valid or exist. Working on it.

File: tests/test-sessions/test-sessions-common/src/test/java/org/eclipse/jetty/server/session/ClientCrossContextSessionTest.java [hunk 1]
```diff
@@ -82,14 +82,16 @@ public void testCrossContextDispatch() throws Exception
                 assertEquals(HttpServletResponse.SC_OK, response.getStatus());
                 String sessionCookie = response.getHeaders().get("Set-Cookie");
                 assertTrue(sessionCookie != null);
-                // Mangle the cookie, replacing Path with $Path, etc.
-                sessionCookie = sessionCookie.replaceFirst("(\\W)(P|p)ath=", "$1\\$Path=");
+                String sessionId = TestServer.extractSessionId(sessionCookie);
 
+                System.err.println("SESSION ID="+sessionId);
```
Comment (gregw) [inline tests/test-sessions/test-sessions-common/src/test/java/org/eclipse/jetty/server/session/ClientCrossContextSessionTest.java]: Don't really need to leave printlns in tests

File: jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java [hunk 2]
```diff
@@ -1627,13 +1628,33 @@ else if (!DispatcherType.REQUEST.equals(baseRequest.getDispatcherType()))
...
+                        if (requestedSessionId == null)
                         {
-                            break;
+                            //no previous id, always accept this one
+                            requestedSessionId = id;
+                            session = s;
+                        }
+                        else
+                        {
+                            if (session == null || !isValid(session))
```
Comment (gregw) [inline jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java]: this could also be an `else if` rather than `else { if`

**Comment by gregw** [comment]
Test failures are:
```
2019-10-08 23:06:51.303:WARN:oejs.HttpChannel:qtp66774422-54: handleException / org.eclipse.jetty.http.BadMessageException: 400: Duplicate valid session cookies: node049ubz48o5dxjpvgpqw6vj5ue0.node0,node049ubz48o5dxjpvgpqw6vj5ue0.node0
```
So looks a bit like a bad test, but we probably should check for identical duplicate session cookies!

**Comment by gregw** [comment]
@joakime can you review this once (if) the tests pass.
I didn't fix the tests that are sending the duplicate identical session cookies... I guess we probably should and then add an explicit test for it...  but maybe get the fix in first.

**Comment by joakime** [comment]
bump, to have jenkins rebuild on merged codebase.

**Pull request merged**